src/runtime/GlobalObjectBuiltinTypedArray.cpp

```diff
@@ -567,11 +567,14 @@ static Value builtinTypedArrayIndexOf(ExecutionState& state, Value thisValue, si
     }
     size_t k = (size_t)doubleK;
 
+    if (O->asTypedArrayObject()->buffer()->isDetachedBuffer()) {
+        return Value(-1);
+    }
```

Comment:
Why did you add this part?
I cannot find any matching point in the spec.

Reply:
True, i updated to the latest standart